txpipe/extensions/twopoint_scia.py

```diff
@@ -0,0 +1,493 @@
+        GAMMAXS = sacc.build_data_type_name('galaxy',['shear','Density'],'xi',subtype ='xs')
+
+        S = sacc.Sacc()
+        if self.config['do_shear_pos'] == True:
+            S2 = sacc.Sacc()
+
+        # We include the n(z) data in the output.
+        # So here we load it in and add it to the data
+        f = self.open_input('shear_photoz_stack')
+
+        # Load the tracer data N(z) from an input file and
+        # copy it to the output, for convenience
+        for i in data['source_list']:
+            z = f['n_of_z/source/z'][:]
+            Nz = f[f'n_of_z/source/bin_{i}'][:]
+            S.add_tracer('NZ', f'source_{i}', z, Nz)
+            if self.config['do_shear_pos'] == True:
+                S2.add_tracer('NZ', f'source_{i}', z, Nz)
+
+
+        f.close()
+
+        # Now build up the collection of data points, adding them all to
+        # the sacc data one by one.
+        comb = []
+        for d in results:
+            # First the tracers and generic tags
+            tracer1 = f'source_{d.i}' #if d.corr_type in [XI, GAMMAT,GAMMATS, ] else f'lens_{d.i}'
+            tracer2 = f'source_{d.j}' #if d.corr_type in [XI, GAMMAT, GAMMATS] else f'lens_{d.j}'
+
+            # We build up the comb list to get the covariance of it later
+            # in the same order as our data points
+            comb.append(d.object)
+
+            theta = np.exp(d.object.meanlogr)
+            npair = d.object.npairs
+            weight = d.object.weight
+
+            # account for double-counting
+            if d.i == d.j:
+                npair = npair/2
+                weight = weight/2
+            # xip / xim is a special case because it has two observables.
+            # the other two are together below
+            if d.corr_type == XI:
+                xip = d.object.xip
+                xim = d.object.xim
+                xiperr = np.sqrt(d.object.varxip)
+                ximerr = np.sqrt(d.object.varxim)
+                n = len(xip)
+                # add all the data points to the sacc
+                for i in range(n):
+                    S.add_data_point(XIP, (tracer1,tracer2), xip[i],
+                        theta=theta[i], error=xiperr[i], npair=npair[i], weight= weight[i])
+                    S.add_data_point(XIM, (tracer1,tracer2), xim[i],
+                        theta=theta[i], error=ximerr[i], npair=npair[i], weight= weight[i])
+            else:
+                xi = d.object.xi
+                err = np.sqrt(d.object.varxi)
+                n = len(xi)
+                for i in range(n):
+                    S.add_data_point(d.corr_type, (tracer1,tracer2), xi[i],
+                        theta=theta[i], error=err[i], npair=npair[i], weight=weight[i])
+
+        # Add the covariance.  There are several different jackknife approaches
+        # available - see the treecorr docs
+        cov = treecorr.estimate_multi_cov(comb, self.config['var_method'])
+        S.add_covariance(cov)
```

Comment:
is there something relevant to specify here for the self-calibration

Reply:
Not really